File: test-suite/src/main/java/example/Book.java
package example;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable // <1>
public class Book {
    private final String title;
    @JsonProperty("qty") // <2>
    private final int quantity;

    @JsonCreator // <3>
    public Book(String title, int quantity) {
        this.title = title;
        this.quantity = quantity;
    }

    public String getTitle() {
        return title;
    }

    public int getQuantity() {
        return quantity;
    }
}

File: doc-examples/example-python/src/main/python/example/Book.py
from dataclasses import dataclass
from typing import Annotated

from com.fasterxml.jackson.annotation import JsonProperty
from micronaut.serde.annotation import Serdeable


@Serdeable  # <1>
@dataclass
class Book:
    title: str
    quantity: Annotated[int, JsonProperty("qty")]  # <2>
